### ISSUE TITLE
Use new fs API

### ISSUE THREAD

**Issue by davidanthoff** [julia-vscode/julia-vscode#820]
See [here](https://code.visualstudio.com/updates/v1_37#_vscodeworkspacefs).

**Comment by DonJayamanne** [comment]
Based on my experience, this is required when dealing with user input or workspace files. As for opening files that are shipped with the extension we can continue using the node FS api.
E.g. if a user were to open a remote julia file, in that instance its best for extension to use the VSC FS Api to read the contents of the julia file (this is just an example, ideally one would use existing text document API).

I.e. IMHO I dont think we should udpate all of the code to use the VSC API (not assuming this is what the original intent was, but would like to ensure anyone picking this up is aware of that) - again this is my opinion.

**Comment by pfitzseb** [comment]
Yeah I think we're doing fairly well in this regard, although there (unfortunately) are some cases where we need to read files outside of the workspace.

**Comment by davidanthoff** [comment]
Great, that makes sense!

The space where I'm actually really confused is how our LS is supposed to handle that. There is lots of code in the LS that reads files from disc, and none of that works when a remote file is opened. I've never understood what the idea for that is in terms of the LSP :) Or to say it another way, I've always thought that the LSP really needed a part where a LS can open files via LSP, rather than using direct file access. But, that is probably for another day.

**Comment by DonJayamanne** [comment]
> s files from disc, and none of that works when a remote file is opened. I've never understood what the idea for that is in

Generally with LSP, they get the contents of the current file. But not necessarily of other remote files.
I.e. if user is editing a remote file, then LSP gets the contents of that file.
